src/main/java/core/basesyntax/operations/SupplyOperation.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax.operations;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class SupplyOperation implements OperationHandler {
+    private final Storage storage;
+
+    public SupplyOperation(Storage storage) {
+        this.storage = storage;
+    }
+
+    @Override
+    public void apply(FruitTransaction transaction) {
+        if (storage == null) {
+            throw new IllegalArgumentException("Storage cannot be null.");
+        }

```

Comment:
Checking if `storage` is `null` in the `apply` method is unnecessary since `storage` is being passed through the constructor and should not be allowed to be `null`. Ensure that the object is properly initialized before being passed to the constructor.